lib/Dialect/ModArith/Transforms/ConvertToMac.cpp

```diff
@@ -0,0 +1,58 @@
+#include "lib/Dialect/ModArith/Transforms/ConvertToMac.h"
+
+#include "lib/Dialect/ModArith/IR/ModArithOps.h"
+#include "llvm/include/llvm/Support/Debug.h"            // from @llvm-project
+#include "mlir/include/mlir/IR/ImplicitLocOpBuilder.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/GreedyPatternRewriteDriver.h"  // from @llvm-project
+
+#define DEBUG_TYPE "mod-arith-mac"
+
+namespace mlir {
+namespace heir {
+namespace mod_arith {
+
+#define GEN_PASS_DEF_CONVERTTOMAC
+#include "lib/Dialect/ModArith/Transforms/Passes.h.inc"
+
+struct FindMac : public OpRewritePattern<mod_arith::AddOp> {
+  FindMac(mlir::MLIRContext *context)
+      : OpRewritePattern<mod_arith::AddOp>(context) {}
+
+  LogicalResult matchAndRewrite(mod_arith::AddOp op,
+                                PatternRewriter &rewriter) const override {
+    ImplicitLocOpBuilder b(op.getLoc(), rewriter);
+
+    auto *parent = op.getLhs().getDefiningOp();
```

Comment:
What stops it from being the Rhs? (Maybe a canonicalization pattern I missed?) You could make a second pattern, or else have something like

`Value otherArg = op.getRhs()`

and have a branch that swaps them if the other parent is a mul.